packages/astro/src/runtime/server/astro-island.ts

```diff
@@ -127,9 +127,30 @@ declare const Astro: {
 						if (!closest?.isSameNode(this)) continue;
 						slots[slot.getAttribute('name') || 'default'] = slot.innerHTML;
 					}
-					const props = this.hasAttribute('props')
-						? JSON.parse(this.getAttribute('props')!, reviver)
-						: {};
+
+					let props: Record<string, unknown>;
+
+					try {
+						props = this.hasAttribute('props')
+							? JSON.parse(this.getAttribute('props')!, reviver)
+							: {};
+					} catch (e) {
+						let componentName: string =
+							this.getAttribute('component-url') || this.Component?.__name || '<unknown>';
```

Comment:
Where's the `__name` coming from? If it's a framework-specific prop, I think we should remove it for now to simplify it.